### PR TITLE
Set the logger explicitly in tests

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of https://github.com/Lightning-AI/lightning/pull/15662

Some tests work based on the assumption that `TensorBoardLogger` is the default logger or that a logger is enabled by default. This PR makes those tests define the logger explicitly

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda

### CODE REVIEW

File: tests/tests_pytorch/checkpointing/test_model_checkpoint.py [hunk 1]
```diff
@@ -975,7 +984,6 @@ def assert_checkpoint_log_dir(idx):
         assert trainer.global_step == epochs * limit_train_batches
         assert trainer.current_epoch == epochs
         assert trainer.fit_loop.epoch_progress.current.processed == epochs
-        assert_checkpoint_log_dir(idx)
```
Comment (awaelchli) [inline tests/tests_pytorch/checkpointing/test_model_checkpoint.py:978]: why this change?

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/tests_pytorch/checkpointing/test_model_checkpoint.py (hunk 1 above)
Comment (carmocca) [reply tests/tests_pytorch/checkpointing/test_model_checkpoint.py]: Reverted!
